// @progress/kendo-inputs-vue-wrapper/dist/es/KendoRangeSlider/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _vue = require('vue');

var Vue = _interopRequireWildcard(_vue);

var _KendoRangeSliderMixin = require('./KendoRangeSliderMixin');

var _KendoRangeSliderMixin2 = _interopRequireDefault(_KendoRangeSliderMixin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
exports.default = {
    name: 'kendo-rangeslider',
    mixins: [_KendoRangeSliderMixin2.default],
    setup: function setup() {
        var v3 = isV3;
        return {
            v3: v3
        };
    },
    render: function render(createElement) {
        var h = gh || createElement;

        if (this.v3) {
            return h('div', [h('input', {
                name: this.selectionStartName
            }), h('input', {
                name: this.selectionEndName
            })]);
        } else {
            return h('div', [h('input', {
                props: {
                    name: this.selectionStartName
                }
            }), h('input', {
                props: {
                    name: this.selectionEndName
                }
            })]);
        }
    }
};
